proxylib: Make xDS node-id configurable.

jrajahalme opened the pull request:
Cilium needs to distinguish between ACKs from the NPDS clients
connecting from Envoy and proxylib. It also needs to separate between
NPDS clients connecting from the host proxy vs. sidecars. To
facilitate this the node-id used by the proxylib's NPDS client needs
to be configurable.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/5704)
<!-- Reviewable:end -->

rlenglet commented on proxylib/proxylib.go:
```diff
@@ -204,16 +204,18 @@ func Close(connectionId uint64) {
 //export OpenModule
 func OpenModule(params [][2]string, debug bool) uint64 {
-	accessLogPath, xdsPath := "", ""
+	var accessLogPath, xdsPath, nodeId string
```
+1

jrajahalme replied:
ditto.

rlenglet commented on proxylib/proxylib/instance.go:
```diff
@@ -62,28 +62,30 @@ var (
 // OpenInstance creates a new instance or finds an existing one with equivalent parameters.
 // returns the instance id.
-func OpenInstance(xdsPath string, newPolicyClient func(path, nodeId string, updater PolicyUpdater) PolicyClient,
+func OpenInstance(nodeId string, xdsPath string, newPolicyClient func(path, nodeId string, updater PolicyUpdater) PolicyClient,
```
+1

jrajahalme replied:
changed.

rlenglet commented on proxylib/proxylib/instance.go:
```diff
@@ -62,28 +62,30 @@ var (
 	mutex.Lock()
 	defer mutex.Unlock()
-
+	
```
go-fmt to remove the extra spaces.

rlenglet reviewed (changes requested):
LGTM after you address the style and formatting issues. Thanks!

jrajahalme committed:
proxylib: Make xDS node-id configurable.

Cilium needs to distinguish between ACKs from the NPDS clients
connecting from Envoy and proxylib. It also needs to separate between
NPDS clients connecting from the host proxy vs. sidecars. To
facilitate this the node-id used by the proxylib's NPDS client needs
to be configurable.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
